### PR TITLE
Add curator and transfer allocatin ownership

### PR DESCRIPTION
**Pull request by Sriep**
Part of issue https://github.com/0chain/0chain/issues/333

Adds a curator to a list of allocation's curators. A curator can transfer allocations owners.
Adds transaction for a curator to trasnfer ownership.

Requires changes to gosdk  https://github.com/0chain/gosdk/pull/121 and zboxcli https://github.com/0chain/zboxcli/pull/47.

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/sc.go [hunk 1]
```diff
@@ -207,6 +207,11 @@ func (sc *StorageSmartContract) Execute(t *transaction.Transaction,
 	case "cancel_allocation":
 		resp, err = sc.cancelAllocationRequest(t, input, balances)
 
+	case "add_curator":
+		resp, err = "", sc.addCurator(t, input, balances)
+	case "transfer_allocation_request":
```
Comment (andrenerd) [inline code/go/0chain.net/smartcontract/storagesc/sc.go:212]: to drop _request suffix. or to add it to all the other cases.

File: code/go/0chain.net/smartcontract/storagesc/writepool.go [hunk 2]
```diff
@@ -299,9 +301,12 @@ func (ssc *StorageSmartContract) getWritePool(clientID datastore.Key,
 	return
 }
 
-func (ssc *StorageSmartContract) createWritePool(t *transaction.Transaction,
-	alloc *StorageAllocation, balances chainState.StateContextI) (err error) {
-
+func (ssc *StorageSmartContract) createWritePool(
+	t *transaction.Transaction,
+	alloc *StorageAllocation,
+	force bool,
```
Comment (andrenerd) [inline code/go/0chain.net/smartcontract/storagesc/writepool.go:307]: not nice. maybe just don't transfer that "undercapitalized" locks and leave them as is.
as a result there will be no need to patch this and `fill` funcs.

**Comment by andrenerd** [comment]
In fact, i've already implemented allocation transfer (and you know it):
https://github.com/0chain/0chain/pull/345

**Comment by Sriep** [comment]
This is a different form of allocation transfer. Only a curator can do the transfer, and no tokens are moved between the original owner and the new one. I do not know if this is in addition to or in place of https://github.com/0chain/0chain/pull/345.

File: code/go/0chain.net/smartcontract/storagesc/sc.go (hunk 1 above)
Comment (Sriep) [reply code/go/0chain.net/smartcontract/storagesc/sc.go]: `finalizeAllocation`, `addBlobber`, `addValidator`, `stakePoolPayInterests`. The request suffix seems to be a minority choice. Also `cancel_allcoation` semes to not match `cancelAllocationRequest`.

File: code/go/0chain.net/smartcontract/storagesc/writepool.go (hunk 2 above)
Comment (Sriep) [reply code/go/0chain.net/smartcontract/storagesc/writepool.go:309]: Yes, still undecided about how to write this; not sure it's possible to not write it in a not nice fashion. This and some unit tests to add is mainly why it is WIP.

**Comment by guruhubb** [comment]
@andrenerd @Sriep this is related to issue #333 and PR #345 but updated, so since @andrenerd is busy with other issues, Pier can go ahead with combining them into one PR

**Review by andrenerd — APPROVED** [review]

**Pull request merged**